apps/emqx/src/emqx_channel.erl

```diff
@@ -1575,6 +1577,13 @@ enrich_client(ConnPkt, Channel = #channel{clientinfo = ClientInfo}) ->
             {error, ReasonCode, Channel#channel{clientinfo = NClientInfo}}
     end.
 
+set_tenant(_ConnPkt, ConnInfo, ClientInfo = #{clientid := ClientId}) ->
```

Comment:
our ambition is to merge this back to master, so we'll need a global configuration to decide
either the SNI info should be used as tenant ID or not.

Reply:
e.g. `tenant_id_from = none | sni`

Reply:
That's exactly what I want too. I will add this configuration item in subsequent commits.